adobe/spectrum-web-components
# refactor: use @floating-ui/dom the successor to @popperjs/core

## Description
- @popperjs/core@3.0 is now @floating-ui/dom, react accordingly...
  - this allows us to police the [container padding rule](https://spectrum.adobe.com/page/popover/#Container-padding) in Spectrum that we'd skipped previously
  - ensure that a reasonable `min-height` is used when choose the "auto" position of an overlay
  - admits that we are currently managing top level scroll (even in `type="modal"`) and ensure that content is positioned appropriately
  - opens the door for more nuanced overlay types by better surfacing the options around this behavior

### TODO
- [x] merge and ship https://github.com/floating-ui/floating-ui/pull/1502 so we can refloor our version without needing `patch-package`.

## Related issue(s)
- fixes #1988 
- fixes #910
- fixes #2016

## Motivation and context
- keep up with the Joneses
- more closely follow the Spectrum overlay spec
- smooth the experience of overlays at the edges

## How has this been tested?
    1. Go [here](https://floating-ui--spectrum-web-components.netlify.app/) _Preferrably in Firefox, but all browser, too_
    2. Scroll the Side Nav
    3. See that it scrolls
    4. Open the "Theme" Picker
    5. Attempt to scroll the Side Nav
    6. See that it doesn't scroll (depending on window size the root might scroll)
    7. Close the Picker
    8. Attempt to scroll the Side Nav
    9. See that it scrolls
    1. Go [here](https://floating-ui--spectrum-web-components.netlify.app/components/picker/#icons) _Specifically in Firefox_
    2. Open the "Choose an action..." Picker
    3. See that the Menu is made available
    4. Scroll the page
    5. See that the Menu follows the Picker button
    1. Go [here](https://floating-ui--spectrum-web-components.netlify.app/storybook/index.html?path=/story/overlay--detached-element)
    2. See that the overlaid content is at least 8px from the edge of the preview iframe
    1. Go [here](https://floating-ui--spectrum-web-components.netlify.app/storybook/index.html?path=/story/picker--custom)
    2. See that the Menu ends 8px before the end of the preview iframe.

## Types of changes

## Checklist
